tunneling/nextjs/pages/api/tunnel.js

```diff
@@ -1,28 +1,37 @@
-import { withSentry } from "@sentry/nextjs";
+import { withSentry, captureException } from "@sentry/nextjs";
 import axios from "axios";
 import * as url from "url";
 
-async function handler(req, res) {
-  const envelope = req.body;
-  const pieces = envelope.split("\n");
-
-  if (!pieces[0]) {
-    return res.status(400).json({ status: "broken_envelope" });
-  }
+// Change host appropriately if you run your own Sentry instance.
+const sentryHost = "sentry.io";
 
-  const header = JSON.parse(pieces[0]);
+// Set knownProjectIds to an array with your Sentry project IDs which you
+// want to accept through this proxy.
+const knownProjectIds = [];
 
-  if (header.dsn) {
+async function handler(req, res) {
+  try {
+    const envelope = req.body;
+    const pieces = envelope.split("\n");
+    
+    const header = JSON.parse(pieces[0]);
+    
     const { host, path } = url.parse(header.dsn);
+    if (host !== sentryHost) {
+      throw new Error(`invalid host: ${host}`);
+    }
+    
     const projectId = path.endsWith("/") ? path.slice(0, -1) : path;
-    const sentryApi = `https://${host}/api/${projectId}/envelope/`;
-
-    const response = await axios.post(sentryApi, envelope);
-
-    return res.status(200).json(response.data);
+    if (!knownProjectIds.includes(projectId)) {
+      throw new Error(`invalid project id: ${projectId}`);
+    }
+    
+    const url = `https://${sentryHost}/api/${projectId}/envelope/`;
+    return await axios.post(url, envelope);
```

Comment:
Is there a reason we're using a third-party library here? Nextjs has a `fetch` polyfill built in: https://nextjs.org/blog/next-9-4#improved-built-in-fetch-support.

Reply:
This is a question to @vladanpaunovic :)

Reply:
No valid reason from me here.
I extracted this example from one of my projects where axios was in already. Feel free to replace it to `fetch`.